src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -283,7 +273,11 @@ private static void LoadMachineStores()
                 }
             }
 
-            s_machineRootStore = rootStore;
+            var rootStorePal = new CollectionBackedStoreProvider(rootStore, cloneOnRead: true);
+            s_machineIntermediateStore = new CollectionBackedStoreProvider(intermedStore, cloneOnRead: true);
+
+            // s_machineRootStore's nullarity is the loaded-state sentinel, so write it with Volatile.
+            Volatile.Write(ref s_machineRootStore, rootStorePal);
```

Comment:
Is this volatile write because we're in a double-checked locking situation?  I'm assuming so, as it looks like we're inside of a lock here.  If that's not the case, and we're in an optimistic concurrency situation where multiple threads may be racing to publish their created store, this should instead be an Interlocked.CompareExchange, with the loser disposing of their created state.